lib/client/api.go

```diff
@@ -535,6 +538,11 @@ func (c *Config) SaveProfile(profileDir string, profileOptions ...ProfileOptions
 	profileDir = FullProfilePath(profileDir)
 	profilePath := path.Join(profileDir, webProxyHost) + ".yaml"
 
+	profileAliasPath := ""
```

Comment:
Proxy alias logic is split in two different places in `SaveProfile` and `SaveTo`. However, `SaveTo` appears to mainly be working with files on disk where as `SaveProfile` is updating configuration.

Moving that configuration logic from `SaveTo` into `SaveProfile` would simply things. Something like the following.

```go
// Set alias path if it differs from the address advertised by Teleport.
aliasPath = ""
if profilePath != profileAliasPath {
    aliasPath = profileAliasPath
}

err := SaveTo(&profile{
    path: profilePath,
    alias: aliasPath,
    options: ...,
})
```

Then `SaveTo` once again only works with files on disk:

```go
// If a alias has been provided, update the symlink to to point to the profile
// that Teleport advertises.
if p.alias != "" {
    if err := os.Remove(profileAliasPath); err != nil {
        log.Warningf("Failed to remove symlink alias: %v", err)
    }

    err := os.Symlink(filepath.Base(filePath), profileAliasPath)
    if err != nil {
        log.Warningf("Failed to create profile alias: %v", err)
    }
}
```